client/src/pages/Login.js

```diff
@@ -278,117 +328,158 @@ const Login = ({ isLoginForm }) => {
                     })}
                     style={{ ...inputStyles, paddingRight: "3.5rem" }}
                   />
                   <VisibilityButton
-                    onClick={toggleConfirmPasswordVisibility}
-                    type={confirmPasswordType}
+                    onClick={togglePasswordVisibility}
+                    type={passwordType}
                   />
-                  {errors.confirmPassword && (
-                    <InputError>{errors.confirmPassword.message}</InputError>
+                  {errors.password && (
+                    <InputError>{errors.password.message}</InputError>
                   )}
                 </InputWrapper>
-              )}
-              <SubmitButton
-                primary="true"
-                disabled={!formState.isValid}
-                onClick={
-                  isLoginForm
-                    ? handleSubmit(onLoginWithEmail)
-                    : handleSubmit(onSignup)
-                }
-              >
-                {isLoginForm ? "Sign In" : "Sign Up"}
-              </SubmitButton>
-            </form>
+                {!isLoginForm && (
+                  <InputWrapper>
+                    <Label
+                      htmlFor="confirmPassword"
+                      style={blockLabelStyles}
+                      label="Confirm Password"
+                    />
+                    <Input
+                      type={confirmPasswordType}
+                      name="confirmPassword"
+                      id="confirmPassword"
+                      className={errors.confirmPassword && "has-error"}
+                      required
+                      placeholder="Confirm password"
+                      ref={register({
+                        maxLength: PASSWORD_MAX_LENGTH,
+                        minLength: PASSWORD_MIN_LENGTH,
+                        required: "Password confirmation is required.",
+                        validate: comparePasswordConfirmation,
+                      })}
+                      style={{ ...inputStyles, paddingRight: "3.5rem" }}
+                    />
+                    <VisibilityButton
+                      onClick={toggleConfirmPasswordVisibility}
+                      type={confirmPasswordType}
+                    />
+                    {errors.confirmPassword && (
+                      <InputError>{errors.confirmPassword.message}</InputError>
+                    )}
+                  </InputWrapper>
+                )}
+                <SubmitButton
+                  primary="true"
+                  disabled={!formState.isValid}
+                  onClick={
+                    isLoginForm
+                      ? handleSubmit(onLoginWithEmail)
+                      : handleSubmit(onSignup)
+                  }
+                >
+                  {isLoginForm ? "Sign In" : "Sign Up"}
+                </SubmitButton>
+              </form>
+            ) : (sentRecoveryLink) ? (
```

Comment:
```suggestion
            ) : (recoveryLink) ? (
```

Reply:
Pushed changes for the fix commit 6e7fefe